app/main.py

```diff
@@ -1,13 +1,55 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    formatted_errors = [
+        {
+            "line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"],
+            "name": error["code"],
+            "source": "flake8"
+        }
+        for error in errors
+    ]
```

Comment:
The logic for formatting errors is repeated in multiple functions. Consider creating a helper function to avoid redundancy and improve maintainability.